cpp/src/join/join_common_utils.hpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/detail/utilities/hash_functions.cuh>
+
+#include <limits>
+#include <memory>
+#include <algorithm>
+#include <numeric>
+
+namespace cudf {
+
+namespace detail {
+
+using output_index_type = ::cudf::size_type;
+constexpr output_index_type MAX_JOIN_SIZE{std::numeric_limits<output_index_type>::max()};
+
+constexpr int DEFAULT_CUDA_BLOCK_SIZE = 128;
```

Comment:
This is too generic a name -- it's not the default for all of CUDA. 
```suggestion
constexpr int DEFAULT_JOIN_BLOCK_SIZE = 128;
```